### PR TITLE
feat(workspace/app): add data source to query authorizations

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:

Add data source to query authorizations.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add data source to query authorizations.
2. add correponding document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.

```
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccDataSourceAppGroupAuthorizations_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccDataSourceAppGroupAuthorizations_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceAppGroupAuthorizations_basic
=== PAUSE TestAccDataSourceAppGroupAuthorizations_basic
=== CONT  TestAccDataSourceAppGroupAuthorizations_basic
--- PASS: TestAccDataSourceAppGroupAuthorizations_basic (86.70s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 86.747s
```

* [x] Documentation updated.
* [x] Schema updated.
* [ ] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

    
  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
    \>>>>>> Paste the screenshot here <<<<<<

### CODE REVIEW

**Comment by dengyali0125** [comment]
LGTM

**Comment by WangLiNaruto** [comment]
LGTM

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccResourceAppGroupAuthorization_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccResourceAppGroupAuthorization_basic -timeout 360m -parallel 4
=== RUN   TestAccResourceAppGroupAuthorization_basic
=== PAUSE TestAccResourceAppGroupAuthorization_basic
=== CONT  TestAccResourceAppGroupAuthorization_basic
--- PASS: TestAccResourceAppGroupAuthorization_basic (50.22s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 50.267s


make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccDataSourceAppGroupAuthorizations_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccDataSourceAppGroupAuthorizations_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceAppGroupAuthorizations_basic
=== PAUSE TestAccDataSourceAppGroupAuthorizations_basic
=== CONT  TestAccDataSourceAppGroupAuthorizations_basic
--- PASS: TestAccDataSourceAppGroupAuthorizations_basic (113.56s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 113.635s

File: docs/data-sources/workspace_app_group_authorizations.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  account      = var.account_name_prefix
+  account_type = "USER"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the resource.
```
Comment (Lance52259) [inline docs/data-sources/workspace_app_group_authorizations.md]:
```suggestion
* `region` - (Optional, String) Specifies the region in which to query the authorizations.
```
resources or authorizations is right

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccDataSourceAppGroupAuthorizations_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccDataSourceAppGroupAuthorizations_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceAppGroupAuthorizations_basic
=== PAUSE TestAccDataSourceAppGroupAuthorizations_basic
=== CONT  TestAccDataSourceAppGroupAuthorizations_basic
--- PASS: TestAccDataSourceAppGroupAuthorizations_basic (82.32s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 82.373s

**Pull request merged**